catalog/plugin/catalog-plugin-metacardbackup/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/MetacardBackupPlugin.java

```diff
@@ -261,4 +285,24 @@ private MetacardTransformer lookupTransformerReference() {
         }
         return null;
     }
+
+    private boolean shouldBackupMetacard(Metacard metacard) {
+        if (backupInvalidMetacards) {
+            return true;
+        } else {
+            Attribute metacardTagsAttr = metacard.getAttribute(Core.METACARD_TAGS);
+            if (metacardTagsAttr != null) {
+                List<Serializable> metacardTags = metacardTagsAttr.getValues();
```

Comment:
This seems like a good use for streams:

metacardTagsAttr.getValues().stream()
  .filter(String.class::isInstance)
  .map(String.class::cast)
  .filter(x -> equalsIgnoreCase(INVALID_TAG)) 
  ...